web/directory/management/commands/parse_coop_csv.py

```diff
@@ -122,6 +124,17 @@ def get_address_pks(file_path, city_pks):
         i=1
         address_pks = dict()
         for row in input_file:
+
+            # New code for testing whether the address can be public
+            try:
+                adrs_pub = row['ent-adrs-pub'].strip().encode("utf-8", 'ignore').decode("utf-8")
+            except KeyError:
+                adrs_pub = row['ent-adrs-pub'].strip().encode("utf-8", 'ignore').decode("utf-8")
```

Comment:
This line is identical to line 130.  To reduce code, can you use the ".get" method when accessing the "row" dictionary?  ".get" will not throw an error if it can't find they key.  See this guy's SO post -- https://stackoverflow.com/questions/46024562/how-do-i-avoid-keyerror-when-working-with-dictionaries